services/inc/check.h

```diff
@@ -15,6 +15,107 @@
  * License along with this library; if not, see <http://www.gnu.org/licenses/>.
  */
 
+/**
+ * @file check.h
+ *
+ * @section check_macros Check macros
+ *
+ * This header file defines a set of macros that simplify error handling in the Device OS code.
+ *
+ * Device OS doesn't use C++ exceptions, and, by convention, its internal and public API functions
+ * return a negative value in case of an error. All error codes are defined in the system_error.h
+ * file.
+ *
+ * Consider the following example:
+ * ```
+ * int foo() {
+ *     void* buffer = malloc(1024);
+ *     if (!buffer) {
+ *         return SYSTEM_ERROR_NO_MEMORY; // -260
+ *     }
+ *     ...
+ *     free(buffer);
+ *     return 0; // No error
+ * }
+ * ```
+ * When invoking such a function, the calling function should properly handle all possible errors.
+ * In the most typical scenario, it simply forwards the error to its own calling function:
+ * ```
+ * int fooBarBaz() {
+ *     int result = foo();
+ *     if (result == 0) {
+ *         result = bar();
+ *         if (result == 0) {
+ *             result = baz();
+ *         }
+ *     }
+ *     return result;
+ * }
+ * ```
+ * When many functions are involved, writing the code like in the above example becomes tedious.
+ * Given that all the functions follow the convention, the code can be rewritten as follows using
+ * the `CHECK()` macro:
+ * ```
+ * int fooBarBaz() {
+ *     // This macro makes fooBarBaz() return SYSTEM_ERROR_NO_MEMORY if foo() fails to allocate its
+ *     // internal buffer
+ *     CHECK(foo());
+ *     CHECK(bar());
+ *     CHECK(baz());
+ *     return 0;
+ * }
+ * ```
+ * The `CHECK()` macro expands to a code that evaluates the expression passed to the macro, and if
```

Comment:
Please refactor this sentence. The meaning of "calling code" vs "function that invokes the macro" were not immediately obvious. I had to read it twice to understand the MACRO forces an early return from the calling function.